CMakeLists.txt

```diff
@@ -63,51 +62,102 @@ set( TRACE_SOURCE_FILES
 
+######### installation ####################################
 
-######### Tests ###########################################
-enable_testing()
+install( TARGETS       ${LIBRARY_NAME}
+         EXPORT        ${PROJECT_NAME}Targets
+         LIBRARY       DESTINATION "${CMAKE_INSTALL_LIBDIR}"                            COMPONENT shared_lib
+         ARCHIVE       DESTINATION "${CMAKE_INSTALL_LIBDIR}"                            COMPONENT static_lib
+         RUNTIME       DESTINATION "${CMAKE_INSTALL_BINDIR}"                            COMPONENT executable
+         PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${LIBRARY_NAME}"        COMPONENT headers)
 
-set( TEST_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR} )
+set( ${PROJECT_NAME}_CMAKE_FILES_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${LIBRARY_NAME}" )
 
-set( TEST_SOURCES
-    tests/main.cpp
-    tests/should_not_throw.cpp
-    tests/should_throw.cpp )
+install( EXPORT      ${PROJECT_NAME}Targets
+         NAMESPACE   ${PROJECT_NAME}::
+         FILE        ${PROJECT_NAME}Targets.cmake
+         DESTINATION "${${PROJECT_NAME}_CMAKE_FILES_DESTINATION}" )
 
-# Executable
-add_executable( test_trace ${TEST_SOURCES} )
-target_link_libraries( test_trace ${LIBRARY_NAME} )
+file( WRITE "${CMAKE_CURRENT_BINARY_DIR}/Config.cmake.in" [[
+@PACKAGE_INIT@
 
-# Properties
-set_target_properties( test_trace PROPERTIES
-    RUNTIME_OUTPUT_DIRECTORY ${TEST_RUNTIME_OUTPUT_DIRECTORY}
-    CXX_STANDARD          17
-    CXX_STANDARD_REQUIRED ON
-    CXX_EXTENSIONS        OFF
-    COMPILE_FLAGS         "${TRACE_COMPILE_FLAGS} ${TRACE_FLAGS_COVERAGE}"
-    LINK_FLAGS            "${TRACE_LINK_FLAGS}    ${TRACE_FLAGS_COVERAGE}" )
+include ( "${CMAKE_CURRENT_LIST_DIR}/TraceTargets.cmake" )
+]] )
+
+include( CMakePackageConfigHelpers )
+
+configure_package_config_file(
+    "${CMAKE_CURRENT_BINARY_DIR}/Config.cmake.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+    INSTALL_DESTINATION "${${PROJECT_NAME}_CMAKE_FILES_DESTINATION}"
+    NO_SET_AND_CHECK_MACRO
+    NO_CHECK_REQUIRED_COMPONENTS_MACRO )
 
-# Add the test
-add_test( NAME              test_trace
-          WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-          COMMAND           test_trace --use-colour=yes )
+write_basic_package_version_file (
+  "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+  VERSION       "${${PROJECT_NAME}_VERSION}"
+  COMPATIBILITY AnyNewerVersion
+)
+
+install( FILES       "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+                     "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+         DESTINATION "${${PROJECT_NAME}_CMAKE_FILES_DESTINATION}" )
+
+export( EXPORT ${PROJECT_NAME}Targets
+        FILE   "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Targets.cmake" )
+###########################################################
+
+
+######### Tests ###########################################
+option( BUILD_TESTING ON )
```

Comment:
Please place `option`s near the top of the CMakeLists.txt, so they are obvious when reading this file.

Something about your changes (I cannot currently see the reason) is also keeping the tests from being built on the CI, which is therefore currently failing.

Reply:
Right. I remember now: I if'ed all the tests because I couldn't compile `catch` due to https://github.com/catchorg/Catch2/issues/2178 . You should probably update the header.

Reply:
I created a corresponding issue #3